webview-ui/src/components/chat/ChatRow.tsx

```diff
@@ -953,6 +953,88 @@ export const ChatRowContent = ({
 											}
 										}
 
+										// Check for rate limit errors (status code 429)
+										const isRateLimitError =
+											apiRequestFailedMessage?.includes("status code 429") ||
+											apiRequestFailedMessage?.toLowerCase().includes("rate limit") ||
+											apiRequestFailedMessage?.toLowerCase().includes("too many requests") ||
+											apiRequestFailedMessage?.toLowerCase().includes("quota exceeded") ||
+											apiRequestFailedMessage?.toLowerCase().includes("resource exhausted")
+
+										if (isRateLimitError) {
+											// Check if current provider is Gemini CLI to show specific message
+											const isGeminiCliProvider = apiConfiguration?.apiProvider === "gemini-cli"
+
+											if (isGeminiCliProvider) {
+												return (
+													<div
+														style={{
+															backgroundColor: "rgba(255, 191, 0, 0.1)",
+															padding: "12px",
+															borderRadius: "4px",
+															border: "1px solid rgba(255, 191, 0, 0.3)",
+														}}>
+														<div
+															style={{
+																display: "flex",
+																alignItems: "center",
+																marginBottom: "8px",
+															}}>
+															<i
+																className="codicon codicon-warning"
+																style={{
+																	marginRight: "8px",
+																	fontSize: "16px",
+																	color: "#FFA500",
+																}}></i>
+															<span
+																style={{
+																	fontWeight: "bold",
+																	color: "#FFA500",
+																}}>
+																Rate Limit Exceeded
+															</span>
+														</div>
```

Comment:
These should be using theme based colors instead of hard coded colors.  This helps with accessibility for those that have trouble with colors.